### ISSUE TITLE
Using the na_ontap_info module fails due to insufficient access with a read-only account

### ISSUE THREAD

**Issue by db0** [ansible-collections/netapp#68]
##### SUMMARY
I'm trying to test that the ontap module works correctly before starting development using it. As I'm not a netapp admin, I've only been given read-only access to their server for initial testing.

Unfortunately using a simple call with na_ontap_info fails due to insufficient write privileges, which makes no sense as I'm not attempting to write anything.

I'm assuming the overall module is testing for write access, even when using just ontap_info

Apologies I am not using the latest collection, but it's a bit too bleeding edge for our env. Didn't know if there's any other repo to post this issue though. If this behaviour is known has been fixed in the ansible collection, please let me know.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
na_ontap_info

##### ANSIBLE VERSION

```paste below
ansible 2.9.1
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/ec/test/app/rundecktest/ansible/brocade/library']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.8 (default, Sep 26 2019, 11:57:09) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]

```

##### CONFIGURATION

```paste below
ANSIBLE_PIPELINING(/etc/ansible/ansible.cfg) = True
ANSIBLE_SSH_ARGS(/etc/ansible/ansible.cfg) = -C -o ControlMaster=auto -o ControlPersist=10s -o LogLevel=QUIET -o PreferredAuthentications=publickey
DEFAULT_MODULE_PATH(env: ANSIBLE_LIBRARY) = ['/ec/test/app/rundecktest/ansible/brocade/library']
DEFAULT_REMOTE_USER(/etc/ansible/ansible.cfg) = root
DEFAULT_TIMEOUT(/etc/ansible/ansible.cfg) = 10
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = True
RETRY_FILES_SAVE_PATH(/etc/ansible/ansible.cfg) = /root/ansible
```

##### OS / ENVIRONMENT

Ansible control node: Red Hat Enterprise Linux Server release 7.8 (Maipo)
Ansible delegation server: Red Hat Enterprise Linux Server release 7.8 (Maipo)

##### STEPS TO REPRODUCE
Run the below playbook, using an AD account with only read access

```yaml
---
- hosts: all
  gather_facts: no
  
  tasks:
    
  - name: Gather facts
    na_ontap_info:
      state: info 
      hostname: 'nas-server'
      username: 'domain\ro-admin'
      password: "{{ NAS_PASS }}"
      https: true 
      validate_certs: false 
      gather_subset: "ontap_system_version"
    register: netapp_facts

  - name: Show facts
    debug: 
      var: netapp_facts

```

##### EXPECTED RESULTS

Receive the ontap version
##### ACTUAL RESULTS
I'm receiving a 13003:Insufficient privileges

Paste below has server/username redacted

```paste below
TASK [Gather facts] ************************************************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: netapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - 13003:Insufficient privileges: user '*******' does not have write access to this resource
fatal: [*******]: FAILED! => {"changed": false, "module_stderr": "Traceback (most recent call last):\n  File \"<stdin>\", line 102, in <module>\n  File \"<stdin>\", line 94, in _ansiballz_main\n  File \"<stdin>\", line 40, in invoke_module\n  File \"/usr/lib64/python3.6/runpy.py\", line 205, in run_module\n    return _run_module_code(code, init_globals, run_name, mod_spec)\n  File \"/usr/lib64/python3.6/runpy.py\", line 96, in _run_module_code\n    mod_name, mod_spec, pkg_name, script_name)\n  File \"/usr/lib64/python3.6/runpy.py\", line 85, in _run_code\n    exec(code, run_globals)\n  File \"/tmp/ansible_na_ontap_info_payload_bc5j08v7/ansible_na_ontap_info_payload.zip/ansible/modules/storage/netapp/na_ontap_info.py\", line 619, in <module>\n  File \"/tmp/ansible_na_ontap_info_payload_bc5j08v7/ansible_na_ontap_info_payload.zip/ansible/modules/storage/netapp/na_ontap_info.py\", line 613, in main\n  File \"/tmp/ansible_na_ontap_info_payload_bc5j08v7/ansible_na_ontap_info_payload.zip/ansible/modules/storage/netapp/na_ontap_info.py\", line 501, in get_all\n  File \"/tmp/ansible_na_ontap_info_payload_bc5j08v7/ansible_na_ontap_info_payload.zip/ansible/module_utils/netapp.py\", line 507, in ems_log_event\n  File \"/usr/local/lib/python3.6/site-packages/netapp_lib/api/zapi/zapi.py\", line 301, in invoke_successfully\n    raise NaApiError(code, msg)\nnetapp_lib.api.zapi.zapi.NaApiError: NetApp API failed. Reason - 13003:Insufficient privileges: user '*******' does not have write access to this resource\n", "module_stdout": "", "msg": "MODULE FAILURE\nSee stdout/stderr for the exact 
```

**Comment by dprts** [comment]
All `na_ontap_*` modules send a log event so `read-only` role is not sufficient and it has to be slightly expanded by adding `generate-autosupport-log`.
```
::> security login role create -role ansible_ro_role -cmddirname DEFAULT -access readonly 
::> security login role create -role ansible_ro_role -cmddirname "event generate-autosupport-log" -access all

::> security login role show -role ansible_ro_role
           Role          Command/                                      Access
Vserver    Name          Directory                               Query Level
---------- ------------- --------- ----------------------------------- --------
MYCLUSTER ansible_ro_role DEFAULT                                  readonly
                                             event generate-autosupport-log           all
```
                                             `

**Comment by db0** [comment]
Thank you, I will try that and report back

**Comment by db0** [comment]
I have added the suggested rights and can confirm this worked fine. 

Therefore I would like to change this issue to a suggestion to update the documentation for the na_ontap_info to mention this requirement. Thanks

**Comment by lonico** [comment]
fixed in #e8a1918e

**Comment by lonico** [comment]
The updated collection is now available in Galaxy.
